Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs

```diff
@@ -402,6 +402,82 @@ public KryptonPaletteCheckButtons ButtonStyles
 
         #endregion
 
+        #region Image Dictionary
+
+        protected override IDictionary<PaletteButtonSpecStyle, Image> ButtonSpecStyleImageDictionary()
+        {
+            IDictionary<PaletteButtonSpecStyle, Image> imageDictionary =
+                new ConcurrentDictionary<PaletteButtonSpecStyle, Image>();
```

Comment:
Does not need to be a concurrent dictionary, as this will be mostly for reading, which is thread safe anyway. Concurrent is for multithreaded Modifications (add / remove)

Reply:
Also every time this is called, it will be creating a new dictionary of images. This is going to be slow and wasteful.

Reply:
Reworked, so the image dictionary is set up in the palettes